packages/ai/core/util/call-chat-api-v2.ts

```diff
@@ -0,0 +1,110 @@
+import { IdGenerator, JSONValue, UseChatOptions } from '../types';
+import { ChatStore } from './chat-store';
+import { processChatResponseV2 } from './process-chat-response-v2';
+import { processChatTextResponseV2 } from './process-chat-text-response-v2';
+
+// use function to allow for mocking in tests:
+const getOriginalFetch = () => fetch;
+
+export async function callChatApiV2({
+  api,
+  body,
+  streamProtocol = 'data',
+  credentials,
+  headers,
+  abortController,
+  restoreMessagesOnFailure,
+  onResponse,
+  onUpdate,
+  onFinish,
+  onToolCall,
+  generateId,
+  fetch = getOriginalFetch(),
+  store,
+  chatId,
```

Comment:
passing in the `store` and the `chatId` makes more sense to me - doesn't this collide with the interfaces from `ui-messages.ts`?